src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.cs

```diff
@@ -526,50 +526,72 @@ public unsafe dynamic AsTimeOnly(dynamic field)
+            if (CurrentIndexingScope.Current == null)
+                throw new InvalidOperationException("Indexing scope was not initialized.");
+            
             return CreateSpatialField(name, ConvertToDouble(lat), ConvertToDouble(lng));
         }
 
-        public IEnumerable<AbstractField> CreateSpatialField(string name, double? lat, double? lng)
+        public dynamic CreateSpatialField(string name, double? lat, double? lng)
         {
+            if (CurrentIndexingScope.Current == null)
+                throw new InvalidOperationException("Indexing scope was not initialized.");
+            
             var spatialField = GetOrCreateSpatialField(name);
 
             return CreateSpatialField(spatialField, lat, lng);
         }
 
-        public IEnumerable<AbstractField> CreateSpatialField(string name, object shapeWkt)
+        public dynamic CreateSpatialField(string name, object shapeWkt)
         {
+            if (CurrentIndexingScope.Current == null)
+                throw new InvalidOperationException("Indexing scope was not initialized.");
+            
             var spatialField = GetOrCreateSpatialField(name);
             return CreateSpatialField(spatialField, shapeWkt);
         }
 
-        internal static IEnumerable<AbstractField> CreateSpatialField(SpatialField spatialField, object lat, object lng)
+        internal static dynamic CreateSpatialField(SpatialField spatialField, object lat, object lng)
         {
+            if (CurrentIndexingScope.Current == null)
+                throw new InvalidOperationException("Indexing scope was not initialized.");
+            
             return CreateSpatialField(spatialField, ConvertToDouble(lat), ConvertToDouble(lng));
         }
 
-        internal static IEnumerable<AbstractField> CreateSpatialField(SpatialField spatialField, double? lat, double? lng)
+        internal static dynamic CreateSpatialField(SpatialField spatialField, double? lat, double? lng)
         {
+            if (CurrentIndexingScope.Current == null)
+                throw new InvalidOperationException("Indexing scope was not initialized.");
+            
             if (lng == null || double.IsNaN(lng.Value))
                 return Enumerable.Empty<AbstractField>();
             if (lat == null || double.IsNaN(lat.Value))
                 return Enumerable.Empty<AbstractField>();
 
             IShape shape = spatialField.GetContext().MakePoint(lng.Value, lat.Value);
-            return spatialField.CreateIndexableFields(shape);
+            return CurrentIndexingScope.Current.Index.SearchEngineType is SearchEngineType.Lucene
+                ? spatialField.LuceneCreateIndexableFields(shape)
+                : spatialField.CoraxCreateIndexableFields(shape);
         }
 
-        internal static IEnumerable<AbstractField> CreateSpatialField(SpatialField spatialField, object shapeWkt)
+        internal static dynamic CreateSpatialField(SpatialField spatialField, object shapeWkt)
```

Comment:
Why are you doing this?

Should probably be `IEnuemrable<object>` at least

Reply:
Oh, changed to `IEnumerable<object>` everywhere.